### ISSUE TITLE
Module detection issues

### ISSUE THREAD

**Issue by jwadolowski** [terraform-linters/tflint#452]
# Version

```
$ terraform -v
Terraform v0.11.14
+ provider.aws v2.28.1
+ provider.null v2.1.2
+ provider.template v2.1.2

Your version of Terraform is out of date! The latest version
is 0.12.9. You can update by downloading from www.terraform.io/downloads.html
 
$ tflint -v
TFLint version 0.11.2
```

# Bug description

I have quite a few modules that look like this:

```hcl
# Custom module instance
module "aet3-core0" {
  source             = "../../../modules/aet3_core"
  ami_id             = "${data.terraform_remote_state.commons.centos7_ami}"
  availability_zone  = "${data.aws_availability_zones.azs.names[0]}"
  security_group_ids = ["${aws_security_group.aet3-core.id}"]
  subnet_id          = "${aws_subnet.aet.*.id[0]}"
  env_name           = "${var.env_name}"
  domain             = "${data.terraform_remote_state.commons.domain}"
  id                 = "0"
}
```

Everything got initialized without issues:

```
$ terraform init -backend-config="profile=whatever"
Initializing modules...
- module.aet3-core0
- module.aet3-worker0
- module.chef_server
- module.core-0
- module.influxdb-0
- module.jmeter-master-0
- module.jmeter-slave-0
- module.jmeter-slave-1
- module.jmeter-slave-2
- module.zabbix_server

Initializing the backend...

Initializing provider plugins...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.aws: version = "~> 2.28"
* provider.null: version = "~> 2.1"
* provider.template: version = "~> 2.1"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

Despite of that fact `tflint` complains about missing modules and incorrectly suggests `terraform init` execution:

```
$ tflint -f json | jq
{
  "issues": [],
  "errors": [
    {
      "message": "aet3_tier.tf:72,1-20: `aet3-core0` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "aet3_tier.tf:88,1-22: `aet3-worker0` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "chef_tier.tf:44,1-21: `chef_server` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "core_tier.tf:119,1-16: `core-0` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "influxdb_tier.tf:58,1-20: `influxdb-0` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "jmeter_tier.tf:44,1-25: `jmeter-master-0` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "jmeter_tier.tf:62,1-24: `jmeter-slave-0` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "jmeter_tier.tf:80,1-24: `jmeter-slave-1` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "jmeter_tier.tf:98,1-24: `jmeter-slave-2` module is not found. Did you run `terraform init`?; "
    },
    {
      "message": "zbx_tier.tf:37,1-23: `zabbix_server` module is not found. Did you run `terraform init`?; "
    }
  ]
}
```

**Comment by wata727** [comment]
Sorry. TFLint only supports Terraform v0.12.x...

**Comment by ingshtrom** [comment]
I don't believe the version is the problem here. I have the same issue with terraform 0.12.x
```
❯ docker run --rm -e TFLINT_LOG=debug -v $(pwd):/data -t wata727/tflint --module
17:14:03 config.go:83: [INFO] Load config: .tflint.hcl
17:14:03 config.go:95: [INFO] Default config file is not found. Ignored
17:14:03 config.go:104: [INFO] Load fallback config: /root/.tflint.hcl
17:14:03 config.go:112: [INFO] Fallback config file is not found. Ignored
17:14:03 config.go:114: [INFO] Use default config
17:14:03 option.go:51: [DEBUG] CLI Options
17:14:03 option.go:52: [DEBUG]   Module: true
17:14:03 option.go:53: [DEBUG]   DeepCheck: false
17:14:03 option.go:54: [DEBUG]   Force: false
17:14:03 option.go:55: [DEBUG]   IgnoreModules: map[string]bool{}
17:14:03 option.go:56: [DEBUG]   EnableRules: []string(nil)
17:14:03 option.go:57: [DEBUG]   DisableRules: []string(nil)
17:14:03 option.go:58: [DEBUG]   Varfiles: []string{}
17:14:03 option.go:59: [DEBUG]   Variables: []string{}
17:14:03 loader.go:55: [INFO] Initialize new loader
17:14:03 loader.go:66: [INFO] Module manifest file found. Initializing...
17:14:03 loader.go:262: [DEBUG] Parsing the module manifest file: {"Modules":[{"Key":"","Source":"","Dir":"."},{"Key":"test","Source":"../../modules/test","Dir":"../../modules/test"}]}
17:14:03 loader.go:80: [INFO] Load configurations under .
17:14:03 loader.go:95: [INFO] Module inspection is enabled. Building a root module with children...
17:14:03 loader.go:244: [DEBUG] Trying to load the module: key=test, version=, dir=../../modules/test
17:14:03 loader.go:102: [ERROR] Failed to load modules: <nil>: Failed to read module directory; Module directory ../../modules/test does not exist or cannot be read.
Failed to load configurations. 1 error(s) occurred:

Error: Failed to read module directory

Module directory ../../modules/test does not exist or cannot be read.
```

I have set up a test repository [here](https://github.com/ingshtrom/terraform-tflint-test-452)

**Comment by wata727** [comment]
Sorry for the late reply, this issue may be fixed by #840.

**Comment by bendrucker** [comment]
In the Docker example above, `../../` would be above the mounted `/data` directory and seems like it would be inaccessible to TFLint. 

The original issue is due to 0.11. 

#840 should fix some possible issues around loading modules as well but I think the fix(es) here are unrelated.

Feel free to open a new issue if this is still a problem and the advice above doesn't help.
